tidy3d/plugins/smatrix/ports/coaxial_lumped.py

```diff
@@ -0,0 +1,381 @@
+
+import pydantic.v1 as pd
+import numpy as np
+
+from typing import List, Tuple
+
+from ....constants import MICROMETER
+from ....components.geometry.base import Box
+from ....components.geometry.utils_2d import increment_float
+from ....components.types import FreqArray, Axis, Coordinate, Direction
+from ....components.base import cached_property
+from ....components.grid.grid import Grid, YeeGrid
+from ....components.lumped_element import CoaxialLumpedResistor
+from ....components.monitor import FieldMonitor
+from ....components.data.sim_data import SimulationData
+from ....components.source import CustomCurrentSource, GaussianPulse
+from ....components.validators import skip_if_fields_missing
+from ....components.data.data_array import FreqDataArray, ScalarFieldDataArray
+from ....components.data.dataset import FieldDataset
+from ....exceptions import SetupError, ValidationError
+
+from ...microwave import VoltageIntegralAxisAligned, CustomCurrentIntegral2D
+
+from .base_lumped import AbstractLumpedPort
+
+
+class CoaxialLumpedPort(AbstractLumpedPort):
+    """Class representing a single coaxial lumped port
+
+    Example
+    -------
+    >>> port1 = CoaxialLumpedPort(center=(0, 0, 0),
+    ...             outer_diameter=4,
+    ...             inner_diameter=1,
+    ...             normal_axis=2,
+    ...             direction="+",
+    ...             name="coax_port_1",
+    ...             impedance=50
+    ...         )
+    """
+
+    center: Coordinate = pd.Field(
+        (0.0, 0.0, 0.0),
+        title="Center",
+        description="Center of object in x, y, and z.",
+        units=MICROMETER,
+    )
+
+    outer_diameter: pd.PositiveFloat = pd.Field(
+        ...,
+        title="Outer Diameter",
+        description="Diameter of the outer coaxial circle.",
+        units=MICROMETER,
+    )
+
+    inner_diameter: pd.PositiveFloat = pd.Field(
+        ...,
+        title="Inner Diameter",
+        description="Diameter of the inner coaxial circle.",
+        units=MICROMETER,
+    )
+
+    normal_axis: Axis = pd.Field(
+        ...,
+        title="Normal Axis",
+        description="Specifies the axis which is normal to the concentric circles.",
+    )
+
+    direction: Direction = pd.Field(
+        ...,
+        title="Direction",
+        description="Specifies which side of the port the inner conductor is extending from. "
+        "This is needed in order to position the path integral, which is used for computing "
+        "conduction current using Ampère's circuital law.",
+    )
+
+    @cached_property
+    def injection_axis(self):
+        """Injection axis of the port."""
+        return self.normal_axis
+
+    @pd.validator("center", always=True)
+    def _center_not_inf(cls, val):
+        """Make sure center is not infinity."""
+        if any(np.isinf(v) for v in val):
+            raise ValidationError("Center can not contain td.inf terms.")
```

Comment:
single quotes for `td.inf` and probably make `center` lowercase and in single quotes as a field name.